### PR TITLE
feat (coupons): Coupons refactoring

### PR DESCRIPTION
**Pull request by lovrocolic**
## Context

This PR includes several coupons upgrades

## Description

- Invoice template is updated in coupons section
- new API endpoint for fetching applied coupons is added (with filters)
- it's allowed now to add multiple coupons to the customer (also different types of coupons)

Already reviewed PRs:
- https://github.com/getlago/lago-api/pull/610
- https://github.com/getlago/lago-api/pull/606
- https://github.com/getlago/lago-api/pull/604

### CODE REVIEW

File: app/models/credit.rb [hunk 1]
```diff
@@ -38,4 +40,17 @@ def item_name
     # TODO: change it depending on invoice template
     credit_note.invoice.number
   end
+
+  def invoice_coupon_display_name
+    return nil if applied_coupon.blank?
+
+    coupon = applied_coupon.coupon
+    suffix = if coupon.percentage?
+               "#{sprintf("%.2f", applied_coupon.percentage_rate)}%"
```
Comment (jdenquin) [inline app/models/credit.rb:49]: Layout/IndentationWidth: Use 2 (not 11) spaces for indentation. (https://rubystyle.guide#spaces-indentation)

Comment (jdenquin) [inline app/models/credit.rb:49]: Style/FormatString: Favor `format` over `sprintf`. (https://rubystyle.guide#sprintf)

Comment (jdenquin) [inline app/models/credit.rb:49]: Style/StringLiteralsInInterpolation: Prefer single-quoted strings inside interpolations.

File: app/models/credit.rb [hunk 2]
```diff
@@ -38,4 +40,17 @@ def item_name
...
+             else
```
Comment (jdenquin) [inline app/models/credit.rb:50]: Layout/ElseAlignment: Align `else` with `suffix`.

File: app/models/credit.rb [hunk 3]
```diff
@@ -38,4 +40,17 @@ def item_name
...
+               applied_coupon.amount.format
+             end
```
Comment (jdenquin) [inline app/models/credit.rb:52]: Layout/EndAlignment: `end` at 52, 13 is not aligned with `suffix = if` at 48, 4.

File: app/controllers/api/v1/applied_coupons_controller.rb [hunk 4]
```diff
@@ -35,6 +56,12 @@ def create_params
           :percentage_rate,
         )
       end
+
+      def valid_status?(status)
+        return false unless status
```
Comment (rsempe) [inline app/controllers/api/v1/applied_coupons_controller.rb:61]: Not sure we need this line, as `AppliedCoupon.statuses.key?(nil)` return `false` ;)

Comment (lovrocolic) [reply app/controllers/api/v1/applied_coupons_controller.rb:61]: nice catch, will remove it now!

**Pull request merged**